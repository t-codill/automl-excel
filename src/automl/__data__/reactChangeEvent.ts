// tslint:disable: no-any
export const reactChangeEvent: React.ChangeEvent<any> = {
    target: {}
} as unknown as React.ChangeEvent<any>;
