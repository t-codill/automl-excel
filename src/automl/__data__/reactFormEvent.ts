// tslint:disable: no-any
export const reactFormEvent: React.FormEvent<any> = {
    preventDefault(): void { return; }
} as unknown as React.FormEvent<any>;
