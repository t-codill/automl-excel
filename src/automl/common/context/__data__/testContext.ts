import { PageNames } from "../../PageNames";
import { Logger } from "../../utils/logger";
import { IWorkspaceContextProps, WorkspaceError } from "../IWorkspaceProps";
import { WorkspaceFlight } from "../WorkspaceFlight";

export function getLogCustomEventSpy(): jest.SpyInstance<ReturnType<Logger["logCustomEvent"]>> {
    const spy = jest.spyOn(Logger.prototype, "logCustomEvent");
    spy.mockImplementation(jest.fn());
    return spy;
}
let pageName: PageNames = PageNames.Unknown;
export const testContext: IWorkspaceContextProps = {
    get pageName(): PageNames { return pageName; },
    get theme(): string { return "light"; },
    discoverUrls: {},
    flight: new WorkspaceFlight(""),
    location: "eastus",
    resourceGroupName: "testResource",
    subscriptionId: "00000000-0000-0000-0000-000000000000",
    logger: new Logger("test"),
    getToken: () => "testToken",
    setPageName: (value: PageNames) => { pageName = value; },
    workspaceName: "testWorkSpace",
    setNavigationBarButtons(): void { return; },
    onError: (err: WorkspaceError) => { throw err; },
    clearErrors: () => { return; },
    setHeader: () => { return; },
    setLoading: () => { return; }
};
