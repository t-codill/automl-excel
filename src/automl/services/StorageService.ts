import { StorageManagementClient, StorageManagementModels } from "@azure/arm-storage";

import moment from "moment";
import { getResourceGroupName } from "../common/utils/getResourceGroupName";
import { IServiceBaseProps } from "./ServiceBase";
import { ServiceBaseArm } from "./ServiceBaseArm";

export class StorageService extends ServiceBaseArm<StorageManagementClient> {
    constructor(props: IServiceBaseProps) {
        super(props, StorageManagementClient);
    }

    public async getAccount(
        resourceGroupName: string,
        accountName: string
    ): Promise<StorageManagementModels.StorageAccount | undefined> {
        return this.send(async (client, abortSignal) => {
            return (client.storageAccounts.getProperties(resourceGroupName, accountName, { abortSignal }));
        });
    }

    public async getContainer(
        resourceGroupName: string,
        accountName: string,
        containerName: string
    ): Promise<StorageManagementModels.BlobContainer | undefined> {
        return this.send(async (client, abortSignal) => {
            return (client.blobContainers.get(resourceGroupName, accountName, containerName, { abortSignal }));
        });
    }

    public async listAccount(): Promise<StorageManagementModels.StorageAccount[] | undefined> {
        return this.send(async (client, abortSignal) => {
            return (client.storageAccounts.list({ abortSignal }));
        });
    }

    public async listContainer(account: StorageManagementModels.StorageAccount): Promise<StorageManagementModels.ListContainerItems | undefined> {
        return this.send(async (client, abortSignal) => {
            const resourceGroupName = getResourceGroupName(account.id);
            if (!account.name) {
                return undefined;
            }
            if (!resourceGroupName) {
                return undefined;
            }
            await client.blobServices.setServiceProperties(resourceGroupName, account.name, {
                cors: {
                    corsRules: [{
                        allowedOrigins: [window.location.origin],
                        allowedMethods: ["OPTIONS", "GET", "PUT"],
                        maxAgeInSeconds: 86400,
                        exposedHeaders: ["*"],
                        allowedHeaders: ["*"],
                    }]
                }
            });
            return (client.blobContainers.list(resourceGroupName, account.name, {
                abortSignal
            }));
        });

    }

    public async getSasToken(account: StorageManagementModels.StorageAccount): Promise<StorageManagementModels.ListAccountSasResponse | undefined> {
        return this.send(async (client, abortSignal) => {
            if (!account.name) {
                return undefined;
            }
            const resourceGroupName = getResourceGroupName(account.id);
            if (!resourceGroupName) {
                return undefined;
            }
            const response = await client.storageAccounts.listAccountSAS(resourceGroupName, account.name, {
                // sdk bug
                permissions: "rdwlacup" as StorageManagementModels.Permissions,
                resourceTypes: "sco" as StorageManagementModels.SignedResourceTypes,
                protocols: "https",
                services: "b",
                sharedAccessExpiryTime:
                    moment()
                        .add(1, "year")
                        .toDate()
            }, {
                    abortSignal
                });
            return response;
        });

    }
}
