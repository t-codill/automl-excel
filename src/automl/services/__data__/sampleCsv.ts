export const sampleCsv = `a,b,c,d,e
1,2,3,4,5
1,2,3,4,5
1,2,3,4,5
1,2,3,4,5
1,2,3,4,5
`;

const largeCsvHeader = Array(50)
    .fill("aaaaa,bbbbb,ccccc,ddddd,eeeee")
    .join(",");

const largeCsvRow = Array(50)
    .fill("11111,22222,33333,44444,55555")
    .join(",");

export const largeCsv = `${largeCsvHeader}
${Array(50)
        .fill(largeCsvRow)
        .join("\n")}
`;
