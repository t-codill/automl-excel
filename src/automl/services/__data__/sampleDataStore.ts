import { AzureMachineLearningDatastoreManagementModels } from "@vienna/datastore";

export const sampleDataStore: Required<AzureMachineLearningDatastoreManagementModels.DataStoreDto> = {
    dataStoreType: "AzureBlob",
    name: "defaultDataStore",
    azureStorageSection: {
        accountKey: "key",
        accountName: "account",
        credentialType: "AccountKey",
        credential: "secret",
        containerName: "defaultContainer",
        isSas: false
    },
    azureDataLakeSection: {},
    azurePostgreSqlSection: {},
    azureSqlDatabaseSection: {},
    hasBeenValidated: true
};
