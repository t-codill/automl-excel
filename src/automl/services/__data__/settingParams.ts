import { AdvancedSetting } from "../AdvancedSetting";

export const settingParams: AdvancedSetting = {
    jobType: "classification",
    column: "1",
    metric: "AUC_weighted",
    experimentTimeoutMinutes: 60,
    maxIteration: 15,
    experimentExitScore: 0.99,
    maxConcurrent: 1,
    maxCores: 2,
    preprocess: true,
    nCrossValidations: 5,
    blacklistAlgos: null,
    validationSize: 0.2,
    grainColumns: null,
    maxHorizon: null,
    timeSeriesColumn: null
};
