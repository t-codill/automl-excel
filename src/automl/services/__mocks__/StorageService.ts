import { StorageManagementModels } from "@azure/arm-storage";
import { sampleStorageAccount } from "../../../__data__/sampleStorageAccount";
import { sampleStorageContainer } from "../../../__data__/sampleStorageContainer";

export class StorageService {

    public async getAccount(): Promise<StorageManagementModels.StorageAccount | undefined> {
        return sampleStorageAccount;
    }

    public async getContainer(): Promise<StorageManagementModels.BlobContainer | undefined> {
        return sampleStorageContainer;
    }

    public async listAccount(): Promise<StorageManagementModels.StorageAccount[] | undefined> {
        return [sampleStorageAccount, sampleStorageAccount];
    }

    public async listContainer(): Promise<StorageManagementModels.ListContainerItems | undefined> {
        return { value: [sampleStorageContainer] };
    }

    public async getSasToken(): Promise<StorageManagementModels.ListAccountSasResponse | undefined> {
        return { accountSasToken: "sampleToken" };

    }

    public dispose(): void {
        return;
    }

    public reset(): void {
        return;
    }
}
