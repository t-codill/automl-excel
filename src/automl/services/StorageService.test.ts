import { sampleStorageAccount } from "../../__data__/sampleStorageAccount";
import { testContext } from "../common/context/__data__/testContext";
import { StorageService } from "./StorageService";

jest.mock("@azure/arm-storage");

const service = new StorageService(testContext);

describe("StorageService", () => {
    it("should getAccount", async () => {
        const result = await service.getAccount("testResourceGroup", "testAccount");
        expect(result)
            .toMatchSnapshot();
    });

    it("should listAccount", async () => {
        const result = await service.listAccount();
        expect(result)
            .toMatchSnapshot();
    });

    it("should getContainer", async () => {
        const result = await service.getContainer("sampleResourceGroup", "sampleAccount", "sampleContainer");
        expect(result)
            .toMatchSnapshot();
    });

    describe("getSasToken", () => {
        it("should not without account name", async () => {
            const result = await service.getSasToken({ ...sampleStorageAccount, name: undefined });
            expect(result)
                .toBeUndefined();
        });

        it("should not without resource group", async () => {
            const result = await service.getSasToken({ ...sampleStorageAccount, id: "invalidId" });
            expect(result)
                .toBeUndefined();
        });

        it("should", async () => {
            const result = await service.getSasToken({ ...sampleStorageAccount });
            expect(result)
                .toMatchSnapshot();
        });
    });

    describe("listContainer", () => {
        it("should not without account name", async () => {
            const result = await service.listContainer({ ...sampleStorageAccount, name: undefined });
            expect(result)
                .toBeUndefined();
        });

        it("should not without resource group", async () => {
            const result = await service.listContainer({ ...sampleStorageAccount, id: "invalidId" });
            expect(result)
                .toBeUndefined();
        });

        it("should", async () => {
            const result = await service.listContainer(sampleStorageAccount);
            expect(result)
                .toMatchSnapshot();
        });
    });
});
