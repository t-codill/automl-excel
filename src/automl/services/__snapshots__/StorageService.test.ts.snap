// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StorageService getSasToken should 1`] = `
Object {
  "_mockParameters": Array [
    "sampleResourceGroup",
    "sampleStorageAccount",
    Object {
      "permissions": "rdwlacup",
      "protocols": "https",
      "resourceTypes": "sco",
      "services": "b",
      "sharedAccessExpiryTime": "1971-01-01T00:00:00.000Z",
    },
    Object {
      "abortSignal": Object {},
    },
  ],
  "accountSasToken": "sampleToken",
}
`;

exports[`StorageService listContainer should 1`] = `
Object {
  "_mockParameters": Array [
    "sampleResourceGroup",
    "sampleStorageAccount",
    Object {
      "abortSignal": Object {},
    },
  ],
  "value": Array [
    Object {
      "_mockParameters": Array [
        "testResourceGroup",
        "testAccount",
        Object {
          "abortSignal": Object {},
        },
      ],
      "id": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/sampleResourceGroup/Microsoft.Storage/storageAccounts/sampleStorageAccounts",
      "location": "eastus",
      "name": "sampleStorageAccount",
      "primaryEndpoints": Object {
        "blob": "https://blob.samplestorageaccount.com",
      },
    },
  ],
}
`;

exports[`StorageService should getAccount 1`] = `
Object {
  "_mockParameters": Array [
    "testResourceGroup",
    "testAccount",
    Object {
      "abortSignal": Object {},
    },
  ],
  "id": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/sampleResourceGroup/Microsoft.Storage/storageAccounts/sampleStorageAccounts",
  "location": "eastus",
  "name": "sampleStorageAccount",
  "primaryEndpoints": Object {
    "blob": "https://blob.samplestorageaccount.com",
  },
}
`;

exports[`StorageService should getContainer 1`] = `
Object {
  "_mockParameters": Array [
    "sampleResourceGroup",
    "sampleAccount",
    "sampleContainer",
    Object {
      "abortSignal": Object {},
    },
  ],
  "name": "sampleContainer",
}
`;

exports[`StorageService should listAccount 1`] = `
Array [
  Object {
    "_mockParameters": Array [
      "testResourceGroup",
      "testAccount",
      Object {
        "abortSignal": Object {},
      },
    ],
    "id": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/sampleResourceGroup/Microsoft.Storage/storageAccounts/sampleStorageAccounts",
    "location": "eastus",
    "name": "sampleStorageAccount",
    "primaryEndpoints": Object {
      "blob": "https://blob.samplestorageaccount.com",
    },
  },
]
`;
